doc/decisions/0_drafts/key_value.md

```diff
@@ -0,0 +1,123 @@
+# `Key` value API
+
+## Problem
+
+A `Key` has a value, that much is clear.
+From the ["`Key` value types" decision](../3_decided/key_value_types.md) we know that all values are opaque byte-sequences of a known length.
+
+But how should the API for accessing and modifying the value of a `Key` look?
+
+## Constraints
+
+## Assumptions
+
+## Considered Alternatives
+
+### Read-only values
+
+A simple way to define an API would be to make everything read-only.
+A `Key` can be created with a certain value and that's it.
+If you want a different value, create a new `Key`.
+
+That would certainly simplify some parts of the implementation, but it would be major inconvenience for the caller.
+Also allocating entirely new `Key`s just to change values, could lead to unnecessary, short-lived heap-allocations.
+
+### Direct access
+
+The next simplest way to define the API, is to give the caller full direct access.
+
+The caller creates a `Key` with a `void *` and a `size_t`.
+Afterwards the caller an access and change them at any time in any way they want to.
+
+Basically, a `Key` would not have ownership of its value.
+It would just have a reference to it.
+
+This solution could easily lead to various lifetime problems.
+By not owning the value, the `Key` cannot be reponsible for freeing it.
```

Comment:
This also completely breaks copy-on-write

Reply:
True, I'll add it to the file.